### PR TITLE
DS-705, DS-706 | Date/calendar filter

### PR DESCRIPTION
**Pull request by ericbakenhus**
# READY FOR REVIEW

# Summary
- Date/calendar filters

# Review By (Date)
- 🤷 

# Criticality
- Low

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch or deploy preview
2. Navigate to [/events/discovery](https://deploy-preview-875--stanford-alumni.netlify.app/events/discovery/)
4. Mess with the date options
5. Verify events are being filtered correctly
    - NOTE: The filters are currently based on local system timezone, but the cards themselves display PT dates/times. Timezone information is being added to the source data to allow for correct card timezone display in a future PR.

# Associated Issues and/or People
- DS-705
- DS-706

### CODE REVIEW

File: src/components/events-discovery/components/Chips/ChipsComponent.jsx [hunk 1]
```diff
@@ -1,9 +1,68 @@
...
+          ...item.refinements.map((refinement) => ({
+            key: `${refinement.attribute}-${refinement.label}`,
+            attribute: refinement.attribute,
+            label: refinement.label,
+            remove: () => item.refine(refinement),
+          })),
+        ];
+      }, []),
+    [items, midnight]
+  );
```
Comment (ericbakenhus) [inline src/components/events-discovery/components/Chips/ChipsComponent.jsx:65]: Display date chips with date info.

File: src/components/events-discovery/components/DateFilter/DateFilter.jsx
Comment (ericbakenhus) [inline src/components/events-discovery/components/DateFilter/DateFilter.jsx:1]: Date filter component. Went full custom refinement handling to allow for the custom option.

File: src/components/events-discovery/components/Drawer/Drawer.jsx [hunk 2]
```diff
@@ -63,7 +63,8 @@ export const Drawer = ({
     return () => {
       window.removeEventListener('keydown', handleKeyDown);
       if (previouslyFocusedElement) {
-        previouslyFocusedElement.focus();
+        // TODO: This is breaking focus for all filters, do something
+        // previouslyFocusedElement.focus();
```
Comment (ericbakenhus) [inline src/components/events-discovery/components/Drawer/Drawer.jsx:67]: Will look into properly fixing this focus management in DS-716.

File: src/components/events-discovery/components/FilterAccordion/FilterAccordion.jsx
Comment (ericbakenhus) [inline src/components/events-discovery/components/FilterAccordion/FilterAccordion.jsx:1]: Generic filter accordion component.

File: src/components/events-discovery/eventsDiscovery.jsx [hunk 3]
```diff
@@ -64,7 +65,39 @@ const EventsDiscovery = () => (
...
+                subject: routeState.subject,
+              },
+              numericMenu: {
+                startTimestamp: routeState.startTimestamp,
+              },
+            },
+          };
+        },
+      },
+    }}
```
Comment (ericbakenhus) [inline src/components/events-discovery/eventsDiscovery.jsx:100]: Making the query params nicer to look at.

**Comment by sherakama** [comment]
I'm good with this and it does what you say it does.

Just so that I write this down, the new custom filter is mostly a wrapper around the `useNumericMenu` hook which allows for the filtering of ranges. You are then dynamically calculating the times and displaying them in the UI. You're not tracking the state of the refinements outside of the core of React Instantsearch which is good.

Thank you for the detail work on the pretty URLs. I very much like that.

**Review by sherakama — APPROVED** [review]

**Comment by sherakama** [comment]
The year drop-down isn't super helpful. That could probably go.

Screenshot 2024-07-05 at 15 11 42

**Pull request merged**